src/current_user.ts

```diff
@@ -549,28 +599,98 @@ export default class CurrentUser {
             });
           },
           error => {
-            this.instance.logger.verbose(
+            this.apiInstance.logger.verbose(
               `Error fetching users with ids ${userIdsToFetch}:`,
               error,
             );
           },
         );
       })
       .catch((error: any) => {
-        this.instance.logger.verbose(
+        this.apiInstance.logger.verbose(
           `Error fetching messages froom room ${room.name}:`,
           error,
         );
         onError(error);
       });
   }
 
+  fetchAttachment(attachmentURL: string): Promise<any> {
+    if (!this.apiInstance.tokenProvider) {
+      return new Promise<any>((resolve, reject) => {
+        reject(new Error('Token provider not set on apiInstance'));
+      });
+    }
+
+    return this.apiInstance.tokenProvider.fetchToken().then((token: string) => {
+      return sendRawRequest({
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+        method: 'GET',
+        url: attachmentURL,
+      }).then((res: any) => {
+        const attachmentPayload = JSON.parse(res);
+        const fetchedAttachment = PayloadDeserializer.createFetchedAttachmentFromPayload(
+          attachmentPayload,
+        );
+
+        return fetchedAttachment;
+      });
+    });
+  }
+
+  private uploadFile(
+    file: any,
+    fileName: string,
+    roomId: number,
+  ): Promise<any> {
+    const data = new FormData();
+    data.append('file', file, fileName);
+
+    return this.filesInstance
+      .request({
+        body: data,
+        method: 'POST',
+        path: `/rooms/${roomId}/files/${fileName}`,
+      })
+      .then((res: any) => {
+        return JSON.parse(res);
+      });
+  }
+
+  private sendMessageWithCompleteOptions(
+    options: CompleteMessageOptions,
+    onSuccess: (messageId: number) => void,
+    onError: (error: any) => void,
+  ) {
+    this.apiInstance
+      .request({
+        json: options,
+        method: 'POST',
+        path: `/rooms/${options.roomId}/messages`,
+      })
+      .then((res: any) => {
+        const messageIdPayload = JSON.parse(res);
+        const messageId = messageIdPayload.message_id;
+        // TODO: Error handling
```

Comment:
is this still TODO? or does the catch below cover it?

Reply:
Very true